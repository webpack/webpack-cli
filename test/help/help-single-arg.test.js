'use strict';

const chalk = require('chalk');
const { run } = require('../utils/test-utils');
const helpHeader = 'The build tool for modern web applications';

describe('single help flag', () => {
    it('respects --color flag as false', () => {
        const { stdout, stderr } = run(__dirname, ['--help', '--color=false'], false);
        const usage = 'webpack [...options] | <command>';
        const example = 'webpack help --flag | <command>';
        chalk.enabled = true;
        chalk.level = 3;
        const orange = chalk.keyword('orange');
        expect(stdout).not.toContain(orange(usage));
        expect(stdout).not.toContain(orange(example));
        expect(stdout).toContain(usage);
        expect(stdout).toContain(example);
        expect(stderr).toHaveLength(0);
    });
    it('outputs help info with command syntax', () => {
        const { stdout, stderr } = run(__dirname, ['help'], false);
        expect(stdout).toContain(helpHeader);
        expect(stderr).toHaveLength(0);
    });

    it('outputs help info with dashed syntax', () => {
        const { stdout, stderr } = run(__dirname, ['--help'], false);
        expect(stdout).toContain(helpHeader);
        expect(stderr).toHaveLength(0);
    });

    it('creates a readable snapshot', () => {
        const { stderr } = run(__dirname, ['--help'], false);

        const serializer = require('jest-serializer-ansi');
        expect.addSnapshotSerializer(serializer);

        expect(stderr).toHaveLength(0);
    });
});
