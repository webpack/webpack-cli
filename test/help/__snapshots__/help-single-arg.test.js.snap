// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single help flag creates a readable snapshot 1`] = `
"
          ⬡                     webpack                     ⬡

		         https://webpack.js.org

		The build tool for modern web applications

		Usage: \`webpack [...options] | <command>\`

		Example: \`webpack help --flag | <command>\`



Available Commands

  init      Initialize a new webpack configuration                 
  migrate   Migrate a configuration to a new version               
  loader    Scaffold a loader repository                           
  plugin    Scaffold a plugin repository                           
  info      Outputs information about your system and dependencies 
  serve     Run the webpack Dev Server                             

Options

  --entry string           The entry point of your application.             
  -c, --config string      Provide path to a webpack configuration file     
  -m, --merge string       Merge a configuration file using webpack-merge   
  --progress               Print compilation progress during build          
  --silent                 Disable any output that webpack makes            
  --help                   Outputs list of supported flags                  
  --defaults               Allow webpack to set defaults aggresively        
  -o, --output string      Output location of the file generated by webpack 
  --plugin string          Load a given plugin                              
  -g, --global string[]    Declares and exposes a global variable           
  -t, --target string      Sets the build target                            
  -w, --watch              Watch for files changes                          
  -h, --hot                Enables Hot Module Replacement                   
  -s, --sourcemap string   Determine source maps to use                     
  --prefetch string        Prefetch this request                            
  -j, --json               Prints result as JSON                            
  --standard               Prints standard output                           
  -d, --dev                Run development build                            
  -p, --prod               Run production build                             
  --mode string            Defines the mode to pass to webpack              
  --version                Get current version                              
  --node-args string[]     NodeJS flags                                     

                  Made with ♥️  by the webpack team "
`;
